src/components/SpiceList.js

```diff
@@ -8,23 +8,36 @@ class SpiceList extends React.Component {
     search: ""
   }
 
+  handleChange = (evt) => {
+    if (evt.type === "checkbox") {
+      this.setState((prevState) => ({
+        fourStarOnly : evt.checked
+      }))
+    } else {
+      this.setState((prevState) => ({
+        search : evt.value
+      }))
+    }
+  }
+
   renderSpices() {
     return this.props.spices
-      .filter(spice => this.state.fourStarOnly ? spice.rating >= 4 : true)
+      .filter(spice => this.state.fourStarOnly ? spice.rating >= 4 : true) //does saying true mean the same as returning the entire object? is it the same saying as saying return everything?
       .filter(spice => spice.notes.toLowerCase().includes(this.state.search.toLowerCase()))
       .map(spice => (
-        <SpiceItem key={spice.id} spice={spice} />
+        <SpiceItem key={spice.title} spice={spice} /> // i changed key to spice.title, the spice.id was giving an unique "key" prop error.
```

Comment:
Good catch! Since we're not using a database, the new spices won't get a new ID.